assets/src/edit-story/components/library/mediaLibrary.js

```diff
@@ -30,82 +44,284 @@ const Title = styled.h3`
 	font-size: 19px;
 	line-height: 20px;
 	line-height: 1.4em;
-	flex: 2 0 0;
+	flex: 3 0 0;
 `;
 
 const Header = styled.div`
 	display: flex;
-	margin: 0px 4px 10px;
+	margin: 0px 0px 25px;
 `;
 
 const Message = styled.div`
-	 color: ${ ( { theme } ) => theme.colors.fg.v1 };
+	color: ${ ( { theme } ) => theme.colors.fg.v1 };
 	font-size: 19px;
 `;
 
+const FilterButtons = styled.div`
+	border-bottom:  2px solid ${ ( { theme } ) => theme.colors.mg.v1 };
+	padding: 18px 0px;
+	margin: 10px 0px 15px;
+`;
+
+const FilterButton = styled.button`
+	border: 0px;
+	background: none;
+	padding: 0px;
+	margin: 0px 28px 0px 0px;
+	color: ${ ( { theme, active } ) => ( active ? theme.colors.fg.v1 : theme.colors.mg.v1 ) };
+	font-weight: ${ ( { active } ) => ( active ? 'bold' : 'normal' ) };
+	font-size: 13px;
+
+`;
+
+const SearchField = styled.div`
+	position: relative;
+`;
+
+const Search = styled.input.attrs( { type: 'text' } )`
+	width: 100%;
+	background: ${ ( { theme } ) => theme.colors.mg.v1 } !important;
+	border-color: ${ ( { theme } ) => theme.colors.mg.v1 } !important;
+	color: ${ ( { theme } ) => theme.colors.mg.v2 } !important;
+	padding: 2px 10px 2px 33px !important;
+	&::placeholder {
+	  color: ${ ( { theme } ) => theme.colors.mg.v2 };
+	}
+`;
+
+const Icon = styled( Dashicon )`
+	position: absolute;
+	top: 8px;
+	left: 10px;
+	fill: ${ ( { theme } ) => theme.colors.mg.v2 };
+`;
+
+const SUPPORTED_IMAGE_TYPES = [
+	'image/png',
+	'image/jpeg',
+	'image/jpg',
+	'image/gif',
+];
+
+const SUPPORTED_VIDEO_TYPES = [
+	'video/mp4',
+];
+
+const FILTERS = [
+	{ filter: '', name: 'All' },
+	{ filter: 'image', name: 'Images' },
+	{ filter: 'video', name: 'Video' },
+];
+
+const DEFAULT_WIDTH = 150;
+
 function MediaLibrary( { onInsert } ) {
 	const {
-		state: { media, isMediaLoading, isMediaLoaded, mediaType },
-		actions: { loadMedia, setIsMediaLoading, setIsMediaLoaded },
+		state: { media, isMediaLoading, isMediaLoaded, mediaType, searchTerm },
+		actions: { loadMedia, setIsMediaLoading, setIsMediaLoaded, setMediaType, setSearchTerm },
 	} = useLibrary();
 
 	useEffect( loadMedia );
 
+	/**
+	 * Check if number is odd or even.
+	 *
+	 * @param {number}n Number
+	 * @return {boolean} Is even.
+	 */
+	const isEven = ( n ) => {
+		return n % 2 === 0;
+	};
+
+	/**
+	 * Generate height based on ratio of original height / width.
+	 *
+	 * @param {number}oWidth Original element width.
```

Comment:
Odd, doesn't the linter catch a space missing between `}` and `oWidth`?

Reply:
I believe the above is valid jsDocs

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3907/commits/18d1cb294e815978f5c8e0597704f70595a7bcba